### PR TITLE
Bug 1282962 – Inconsistencies with visual cues in private browsing with top tabs

### PR DESCRIPTION
**Pull request by varkor**
There is now a different tab background theme for private- / normal-browsing.
The private browsing button on the top tabs now has the same interaction as the private browsing button in the tab tray.
Overall, this makes the entire experience feel more unified.

### CODE REVIEW

**Comment by tylerlacroix** [comment]
Looking good, I love the purple background on the private mode button. However I have a few thoughts.

- The background in normal mode is very light. I would check with tecgirl but it no longer matches the doc https://docs.google.com/document/d/1fSkzdh-gqmsP0RPRmA1J8N95znt-7Rn9msCYY5rQWpg/edit (I personally prefer the darker background in the doc)
- Since both the tab number and private mode buttons are purple in private mode, it may be worth changing the plus too. (Maybe not)

Also a few little bugs
1. When closing the last tab in normal mode, the private mode button flashes purple.
2. When switching to private for the first time there is a silver outline on the tab when switching to private mode for the first time.
3. There is a squared block when creating a new tab.

[1]
jul-25-2016 14-50-42

[2] 
screen shot 2016-07-25 at 2 37 51 pm

[3] 
giphy

File: Client/Frontend/Browser/TabTrayButtonExtensions.swift [hunk 1]
```diff
@@ -5,20 +5,32 @@
...
 
 extension UIButton {
-    static func privateModeButton() -> UIButton {
-        let privateTab = UIButton()
-        privateTab.setImage(UIImage.templateImageNamed("menu-NewPrivateTab-pbm"), forState: .Normal)
-        privateTab.tintColor = UIColor(white: 0.9, alpha: 1)
-        privateTab.setImage(UIImage(named: "menu-NewPrivateTab-pbm"), forState: .Highlighted)
-        privateTab.accessibilityLabel = NSLocalizedString("Private Tab", comment: "Accessibility label for the Private Tab button in the tab toolbar.")
-        return privateTab
+    class PrivateModeButton: ToggleButton {
```
Comment (sleroux) [inline Client/Frontend/Browser/TabTrayButtonExtensions.swift]: Nit: No need to declare this subclass inside the scope of the extension. I would move this outside.

**Comment by varkor** [comment]
The New Tab button colouring is a good point! It does look better in purple too.

I've fixed the first two bugs you noticed, but I can't reproduce the third one. I'm not sure if this is just because it was fixed as a by-product of fixing the other two or not — does it still occur for you, and if so: does it happen specifically in certain circumstances?

Yeah, maybe the background tabs are too light now — I thought it was nice to differentiate the two modes, but it'd be better to see what works best from more users' perspectives. It'll be easy to remove that change, or adapt it, if need be.

**Comment by tylerlacroix** [comment]
@varkor This is looking amazing! Great job! Yep it looks like the 2 bugs are fixed. :)  As for the last one, it seems like it happens when I switch in and out of private then open a new tab
jul-27-2016 22-14-22

**Comment by varkor** [comment]
Ah, got it! Fixed that one, too!

**Comment by tylerlacroix** [comment]
@varkor Perfect! Thanks, I can confirm all the bugs are fixed! LTGM! :)

**Comment by varkor** [comment]
Note to self:
- [ ] Tab indicator button should dim when there are popovers, in top tab mode.
- [ ] Menu button popover triangle in private mode should be coloured correctly.

**Comment by thebnich** [comment]
Filed [bug 1301787](https://bugzilla.mozilla.org/show_bug.cgi?id=1301787) for the menu arrow issue.

**Pull request merged**